packages/core/src/components/popover-core/popover-core.tsx

```diff
@@ -225,15 +232,21 @@ export class TdsPopoverCore {
     this.popperInstance?.destroy();
   }
 
-  componentDidLoad() {
+  connectedCallback() {
+    if ((this.selector === undefined && this.referenceEl === undefined) || this.disableLogic) {
+      this.disableLogicState = true;
+      console.warn('TDS-POPOVER-CORE: Popover internal logic disabled.');
+      return;
+    }
```

Comment:
@mJarsater  - That is a good approach and something we could do for other components. Over here I am on trail of what are you saying - if a user has provided neither selector nor reference, then component logic goes down. A recommendation was to use an additional prop for those users that want to be 100% sure, aka setting it explicitly. 

The thing is that Popover core is used by three other components so I have to check how much can be done with other props before making braking change, but overall I like idea of controlled and uncontrolled behaviours of component.

Reply:
I think what I am aiming at is that a prop for disabling features in not very intuitive (to me atleast). I think it's a better approach to try and follow coding conventions and implement controlled and uncontrolled states. As far as I can tell this would be non-breaking. 

```
connectedCallback() {
    if ((this.selector === undefined && this.referenceEl === undefined) || this.show !== undefined) {
      return;
    }
```

Reply:
I checked the code and did some changes:

1. Currently, we can not put the `show` prop as a condition that enables a controlled state - aka makes component escape initialization. The reason for that is that other components depend on it. Header's Dropdown and App Launcher break as they need to do initialization to place themselves at the right spot. 

2. I have added the `defaultOpen` prop, which was a good idea. This way, users can put components to be initially visible, without losing closing on click outside etc. There is a rule that `open` prop will take precedence if set.

3. I have added a bit of text in the readme to explain the controlled and uncontrolled state of these components, however, I am not sure if users will have many benefits from the controlled state as in that case, I am certain that position won't work, you need to provide selector/referenceEl, otherwise, the component will be lost in woods. Aka - selector and reference are needed to make this component work as expected. Therefor I think that controlling ´open´ prop is the most we can do for now so I did update description of prop of Popover Canvas and Popover Menu to clarify it better. Tooltip has its ways of handling it. 

Good discussion with you Mr. @mJarsater  ;)